sky/backends/cloud_vm_ray_backend.py

```diff
@@ -2498,7 +2497,6 @@ def is_provided_ips_valid(
         self.stable_internal_external_ips = stable_internal_external_ips
 
     @context_utils.cancellation_guard
-    @annotations.lru_cache(scope='global')
```

Comment:
Please also add a comment here telling the future developer to avoid adding cache here. : )